src/main/java/com/cloudbees/jenkins/support/configfiles/OtherConfigFilesComponent.java

```diff
@@ -0,0 +1,69 @@
+package com.cloudbees.jenkins.support.configfiles;
+
+import com.cloudbees.jenkins.support.api.Component;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.FileContent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.Collections;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Support component for adding xml files to the support bundle.
+ */
+@Extension
+public class OtherConfigFilesComponent extends Component {
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Jenkins.ADMINISTER);
+    }
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Other Jenkins Configuration Files";
+    }
+
+    @Override
+    public void addContents(@NonNull Container container) {
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins != null) {
+            File dir = jenkins.getRootDir();
+            File[] files = dir.listFiles(new FilenameFilter() {
+                public boolean accept(File dir, String name) {
+                    //all xml files but credentials.xml
+                    return name.toLowerCase().endsWith(".xml") && !name.equals("credentials.xml");
+                }
+            });
+            if (files != null) {
```

Comment:
Reading the javadoc, I don't think that code path should ever yield a null `files`. That would basically mean ` jenkins.getRootDir();` returns garbage, which seems unlikely.

Maybe we want to reverse the logic to log a warning (or even an ERROR) here if this happens, or even throw an exception?